lib/honeybadger/util/http.rb

```diff
@@ -37,8 +37,8 @@ def initialize(config)
         @config = config
       end
 
-      def get(endpoint)
-        response = http_connection.get(endpoint)
+      def get(endpoint, headers = nil)
+        response = http_connection.get(endpoint, http_headers(headers))
```

Comment:
This get method is used when performing a check in here:

https://github.com/honeybadger-io/honeybadger-ruby/blob/aabc35a1612c420881aa2c77bd3f624fd3af8c89/lib/honeybadger/backend/server.rb#L41-L50

This change means that these headers will be included with those requests (which is a change in existing behavior):

https://github.com/honeybadger-io/honeybadger-ruby/blob/aabc35a1612c420881aa2c77bd3f624fd3af8c89/lib/honeybadger/util/http.rb#L17-L22

and:

https://github.com/honeybadger-io/honeybadger-ruby/blob/aabc35a1612c420881aa2c77bd3f624fd3af8c89/lib/honeybadger/util/http.rb#L63

This doesn't seem desirable to me; for example, the [check-ins API](https://docs.honeybadger.io/api/reporting-check-ins/) does not use JSON, there is no compression happening, etc.

Thoughts @stympy @subzero10?

Reply:
🤔 I think this was added to be able to make HTTP GET for the `app_http.rb` class, right @halfbyte ? If that's the case, can we:
- create a `base.rb` class with some shared functionality (i.e. `setup_http_connection`)
- create `app_http.rb` and `api_http.rb` with different definitions for HTTP GET

Reply:
I'll just move my modified `#get` to `AppHTTP`, that should fix this. I didn't think of the consequences, really, thanks @joshuap for catching that.

Reply:
I think that works for now. We might need to refactor the http client helpers if we add to them in the future.